File: python/easy/1108_Defanging_an_IP_Address.py
class Solution:
    # 28ms, faster than 76.51% of Python3 online submissions for Defanging an IP Address.
    # 14.3 MB, less than 32.32% of Python3 online submissions for Defanging an IP Address.
    def defangIPaddr(self, address: str) -> str:
        ans = address.split(".")
        return "[.]".join(ans)

    # 32 ms, faster than 43.45% of Python3 online submissions for Defanging an IP Address
    # 14.2 MB, less than 63.49% of Python3 online submissions for Defanging an IP Address
    def defangIPaddr(self, address: str) -> str:
        ans = ""
        for char in address:
            if char != ".":
                ans += char
            else:
                ans += "[.]"
        return ans

    # 32 ms, faster than 43.45% of Python3 online submissions for Defanging an IP Address.
    # 14.2 MB, less than 32.32% of Python3 online submissions for Defanging an IP Address
    def defangIPaddr(self, address: str) -> str:
        return address.replace(".", "[.]")


if __name__ == "__main__":
    so = Solution()
    print(so.defangIPaddr( "255.100.50.0"))

File: c/easy/1108_Defanging_an_IP_Address.c
#include <stdio.h>
#include <stdlib.h>
#include<string.h>

char * defangIPaddr(char *);

int main(void){
    char ip_addr[] = "1.1.1.1";
    printf("%s\n", defangIPaddr(ip_addr));
    return 0;
}

// 0 ms, faster than 100.00% of C online submissions for Defanging an IP Address.
// 5.5 MB, less than 72.20% of C online submissions for Defanging an IP Address.
char * defangIPaddr(char * address){
    char * ip = (char *) malloc(sizeof(char) * (strlen(address)+7));
    int i=0;
    int j=0;
    while (address[i] != '\0'){
        if (address[i] == '.'){
            ip[j] = '['; 
            ip[++j] = '.'; 
            ip[++j] = ']'; 
        }
        else{
            ip[j] = address[i];
        }
            j++;
        i++;
    }
    ip[j] = '\0';
    return ip;
}